docs/Inside_Arbitrum_Nitro/Inside_Arbitrum_Nitro.md

```diff
@@ -0,0 +1,527 @@
+
+![seq-then-exec](seq-then-exec.png)
+
+Let's follow a user's transaction through this process.
+
+First, the user creates a transaction, uses their wallet to sign it, and sends it to the Nitro chain's sequencer. The sequencer's job, as its name implies, is to take the arriving transactions, put them into an ordered sequence, and publish that sequence.
+
+Once the transactions are sequenced, they are run through the *state transition function*, one by one, in order. The state transition function takes as input the current state of the chain (account balances, contract code, and so on), along with the next transaction. It updates the state and sometimes emits a new Layer 2 block on the Nitro chain.
+
+Because the protocol doesn't trust the sequencer not to put garbage into its sequence, the state transition function will detect and discard any invalid transactions in the sequence. A well-behaved sequencer will filter out invalid transactions so the state transition function never sees them--and this reduces cost and therefore keeps transactions fees low--but Nitro will still work correctly no matter what the sequencer puts into its feed. (Transactions in the feed are signed by their senders, so the sequencer can't create forged transactions.)
+
+The state transition function is deterministic, which means that its behavior depends only on the current state and the contents of the next transaction--and nothing else. Because of this determinism, the result of a transaction T will depend only on the genesis state of the chain, the transactions before T in the sequence, and T itself. 
+
+It follows that anyone who knows the transaction sequence can compute the state transition function for themselves--and all honest parties who do this are guaranteed to get identical results. This is the normal way that Nitro nodes operate: get the transaction sequence, and run the state transition function locally. No consensus mechanism is needed for this.
+
+### How the Sequencer Publishes the Sequence
+
+So how do nodes get the sequence? The sequencer publishes it in two ways: a real-time feed, and batches posted on L1 Ethereum.
+
+The real-time feed is published by the sequencer so that anyone who subscribes to the feed receives instant notification of each transaction as it is sequenced. Nitro nodes can subscribe to the feed directly from the sequencer, or through a relay that forwards the feed. The feed represents the sequencer's promise that it will record transactions in a particular order. If the sequencer is honest and doesn't have a long downtime, this promise will be kept. So anyone who trusts the sequencer to keep it promises can rely on the feed to get instant information about the transaction sequence--and they can run the sequenced transactions through the state transition function to learn the results of each transaction immediately. This is "soft finality" for transactions; it's "soft" because it depends on the sequencer keeping its promises.
```

Comment:
```suggestion
The real-time feed is published by the sequencer so that anyone who subscribes to the feed receives instant notifications of each transaction as it is sequenced. Nitro nodes can subscribe to the feed directly from the sequencer, or through a relay that forwards the feed. The feed represents the sequencer's promise that it will record transactions in a particular order. If the sequencer is honest and doesn't have a long downtime, this promise will be kept. So anyone who trusts the sequencer to keep it promises can rely on the feed to get instant information about the transaction sequence--and they can run the sequenced transactions through the state transition function to learn the results of each transaction immediately. This is "soft finality" for transactions; it's "soft" because it depends on the sequencer keeping its promises.
```